### PR TITLE
Only cancel new same_content run

### PR DESCRIPTION
**Pull request by hugoh**
Don't cancel _all_ same_content runs.

Fix to issue #90 (untested, sorry).

### CODE REVIEW

**Comment by fkirc** [comment]
Thanks for submitting the PR.
The logic looks good, but it leads to a conflict if people actually want to skip an older run.
Skipping an older run is a real use case because of https://github.com/fkirc/skip-duplicate-actions/issues/69 and https://github.com/fkirc/skip-duplicate-actions/issues/34.
If I was merging this PR as is, then it could break those valid use cases.
Therefore, we will probably need yet another config-option to make this a non-breaking change.

**Comment by hugoh** [comment]
Hmmm, maybe I'm not fully understanding those issues, but:
* #69 - if there is a newer run, wouldn't it have different content? And if it didn't, then wouldn't the old and new runs completely identical and hence skipping the new or the old would basically not matter?
* #34 - this seems like it should have its own option.

Would this basically mean that, for `same_content`, we'd want an additional flag? Something like `concurrent_skip_preferred`, which could have the following values:
* `oldest`: this is the logic implemented in #112 above
* `push` or `pull_request`: this would be to address the use case from #34 

You obviously have a better understanding of this whole ecosystem of options here, @fkirc.

**Comment by fkirc** [comment]
> Hmmm, maybe I'm not fully understanding those issues, but:
> 
> * #69 - if there is a newer run, wouldn't it have different content? And if it didn't, then wouldn't the old and new runs completely identical and hence skipping the new or the old would basically not matter?

Yes, #69 isn't only about same-content-runs. But there might be cases where the ordering matters.

> * #34 - this seems like it should have its own option.

Perhaps it should, but right now, people are using #34 with a combination of `same_content` and `do_not_skip`, and I would prefer to not break it.

> Would this basically mean that, for `same_content`, we'd want an additional flag? Something like `concurrent_skip_preferred`, which could have the following values:
> 
> * `oldest`: this is the logic implemented in #112 above
> * `push` or `pull_request`: this would be to address the use case from #34

There are many different ways to go forward, but I would propose the following solution:
- Add a new option `concurrent_skipping: same_content_newer` or something similar.
- Implement the new behavior only for the new option.

**Comment by hugoh** [comment]
I pushed another round of changes to implement the `concurrent_skipping: same_content_newer` you suggested.

File: src/index.ts [hunk 1]
```diff
@@ -227,11 +228,19 @@ function detectConcurrentRuns(context: WRunContext) {
...
+  } else if (context.concurrentSkipping === "same_content" || context.concurrentSkipping === "same_content_newer") {
     const concurrentDuplicate = concurrentRuns.find((run) => run.treeHash === context.currentRun.treeHash);
     if (concurrentDuplicate) {
-      core.info(`Skip execution because the exact same files are concurrently checked in ${concurrentDuplicate.html_url}`);
-      exitSuccess({ shouldSkip: true });
+      if (context.concurrentSkipping === "same_content") {
+        core.info(`Skip execution because the exact same files are concurrently checked in ${concurrentDuplicate.html_url}`);
+        exitSuccess({ shouldSkip: true });
+      } else if (context.concurrentSkipping === "same_content_newer") {
+        const concurrentIsOlder = concurrentRuns.find((run) => new Date(run.createdAt).getTime() < new Date(context.currentRun.createdAt).getTime());
```
Comment (fkirc) [inline src/index.ts:238]: I am not so sure about this `concurrentIsOlder`. You are checking whether there exists an older run, but you are not checking whether this older run has the same content (the same treeHash)?

**Review by fkirc — APPROVED** [review]
Well done, thanks for updating the docs as well :+1:

**Pull request merged**